### PR TITLE
Adds estimations of damage values when examining items

### PR DESCRIPTION
**Pull request by cacogen**
## About The Pull Request

<details>
  <summary>SCREENSHOTS (click to expand)</summary>
  

</details>

Initially started as wanting to do #58845 a bit differently but after that got killed it evolved into doing it in a way that might actually get merged.

This gives you an estimation of the damage an item will do when examining it. These estimations are based on the system currently used by floor tiles to describe their throwforce.

It gives you force, throwforce, stamina damage, block chance and armour piercing amount for items. It shows damage, damage type and stamina damage for the chambered projectile in guns and projectiles themselves. These only show up if the damage is noteworthy, though.

Leaves out pellet amount and caliber for projectiles and damage type for items from the original PR. I think these are obvious or unnecessary. Doesn't show all attack types for energy weapons either, only the currently selected one.

I think everything else is consistent with the original PR. I wanted to do this myself a long time ago but lacked the knowledge.

Assuming this doesn't get closed there's room for this to be fleshed out with more items stats that affect combat in future. Hopefully this will shed light on mechanics that have never been explained in-game.

Also makes objects that cause stamina damage use the same var and makes it an object proc like force.

## Why It's Good For The Game

This makes notable combat items obvious to everybody. It stops codediving or testing weapons with a health analyser from being necessary to equip yourself properly for combat. It also stops you grabbing something that doesn't do much or any damage because it looks like it should.

Adds armour penetration and block chance values to items. These are obscure stats you currently have no way of seeing on items in-game. Maybe they'll be added to more things if there's a way to see them without looking at the code.

## Changelog
:cl: cacogen
balance: You can examine items to see estimations of their damage values if they're notable
/:cl:

### CODE REVIEW

**Comment by Colovorat** [comment]
Maybe image of it? Would be pog

**Comment by cacogen** [comment]
@Colovorat there you go

**Comment by MrDoomBringer** [comment]
Having vauge "This weapon does great brute damage" doesn't solve the problem of dabbing on the wiki. Players are just going to ignore this.

If this is done at all, the specific values should be given, and they should be attainable from an easy-to-get tool (say, an analyzer), to make it more available to new players. This doesn't overlap with @YakumoChen's tooltip force descriptions as they would serve different niches.

Still up to @Mothblocks though, it looks like they blocked the last PR and I'm not certain this one holds more merit to them at this stage.

**Comment by Colovorat** [comment]
> @Colovorat there you go

Either me stoopid or i don't see it

**Comment by cacogen** [comment]
Look at the screenshots. Players aren't going to ignore this. It certainly doesn't need specific values to be useful. They're too gamey anyway, especially considering damage output doesn't vary. An analyser would be a clunky way to pick out a weapon. Limiting it to sec makes sense as they're the ones who should have the combat advantage and they need incentives for people to pick the job. I don't think new players need to know which objects crit fastest. Mothblocks could change his mind, given this addresses some of his concerns from the previous PR.

@Colovorat 

You have to click that thing to expand them

**Comment by Colovorat** [comment]
Oh now i see at the top

**Comment by itseasytosee** [comment]
We ought to just delete the wiki, who uses it anyways?

**Comment by Time-Green** [comment]
> We ought to just delete the wiki, who uses it anyways?

Wiki is cool, don't be a butthead

**Comment by itseasytosee** [comment]
If you don't have all the chemistry recipes and virologist symptoms memorized why even play the game?

**Comment by Nexofas** [comment]
If you didnt memorize all xenobio slimes and crossbreeds , all machine hacking procedures, all surgery steps, all weapon damage stats, all armor stats, all stamina damage values why play the game?

**Comment by bluetshirtguy** [comment]
I honestly think this is better to add to the science goggles than it is the sec huds

**Comment by Mothblocks** [comment]
> Having vauge "This weapon does great brute damage" doesn't solve the problem of dabbing on the wiki. 

I don't like the wiki from a designer standpoint, but flat out saying the exact stat values is a bit much for me. Part of this game's charm is the amount of intricate stuff you learn over time. Imagine if we told you ever interaction you could do with an item to other items just by examining it, for instance.

**Comment by Mothblocks** [comment]
This one is, alright.

**Comment by necromanceranne** [comment]
> > Having vauge "This weapon does great brute damage" doesn't solve the problem of dabbing on the wiki.
> 
> I don't like the wiki from a designer standpoint, but flat out saying the exact stat values is a bit much for me. Part of this game's charm is the amount of intricate stuff you learn over time. Imagine if we told you ever interaction you could do with an item to other items just by examining it, for instance.

The problem with a PR like this also comes from a lack of standardization, and that the miniscule value differences we do have between items of a similar class do have some impacts on their capabilities while not neatly fitting into categories or immediately apparent without trial and error/code diving.

An example might be something like this PR (https://github.com/tgstation/tgstation/pull/29995), where the difference of a value of 1 was enough to justify this change. This interaction I don't believe is possible anymore, but it is the kind of mechanical oddities that are still swimming around in the game, and most of it was discovered either by looking at the raw code or experimentation in the sandbox.

This basically makes any statistical information provided by an examination, if obfuscated, pretty pointless. There are too many niche mechanics that hinge on the difference of a +1, and you would already need a broad understanding of the game as a whole to truly appreciate what any of that information actually means, which is only possible if you start collaborating with others and start reading the code itself.

**Comment by cacogen** [comment]
I thought about it and removed the SecHUD requirement to see text. Other item stats that show up on examine (like embed chance, for example) aren't obfuscated by HUDs. 

This makes strong items obvious to everybody, which stops codedivers from having an advantage. It also stops you grabbing something that doesn't do much or any damage because it looked like it should. 

It also adds armour penetration and block chance, like the other PR. These are obscure stats you have no way of recognising on items in-game. Maybe they'll get added to more things if there's a way to see them without looking at the code.

**Comment by cacogen** [comment]
@necromanceranne 

I agree it's hard to pick out a weapon with a damage output higher than the threshold of things like windows and airlocks if everything is described in ranges. 

But I still think giving out the exact damage number is too gamey. Ideally, it would slightly vary each attack.

Short of adding a description to items saying whether they can damage windows or airlocks (seems a bit too specific) I think leaving people to experiment, learn through word of mouth or code dive is the only option. 

A solution might be to have the damage ranges coincide with thresholds of common items like windows and airlocks but I don't know how you'd maintain this over time. 

If it isn't already, it should be obvious when attacking something is doing nothing (without having to examine it for a damage message). This would speed up experimentation.

File: code/modules/projectiles/ammunition/_ammunition.dm [hunk 1]
```diff
@@ -70,7 +70,7 @@
 			damage_rundown_message += "[return_blows2crit_rating_string(loaded_projectile.damage*pellets)] [loaded_projectile.damage_type] damage"
 
 		if(LAZYLEN(damage_rundown_message))
-			var/start_of_message = "<span class='danger'>Firing it would do "
+			var/start_of_message = "<span class='danger'>Firing it[pellets > 1 ? "s [pellets] pellets" : ""] would do "
```
Comment (Mothblocks) [inline code/modules/projectiles/ammunition/_ammunition.dm:73]: You can just do `\s`

**Comment by Mothblocks** [comment]
I like the other PR as well, I'm going to give a good bit of thought between this and #58865.

File: code/modules/projectiles/ammunition/_ammunition.dm (hunk 1 above)
Comment (cacogen) [reply code/modules/projectiles/ammunition/_ammunition.dm:73]: The damage is based on all the pellets hitting, I'm trying to allude to that here.

Comment (Mothblocks) [reply code/modules/projectiles/ammunition/_ammunition.dm:73]: I see, nevermind.

**Comment by Mothblocks** [comment]
Okay, I talked it out with some people I trust and I'm going to choose the other PR over this, I'll explain why over there.

I'm going to mark this DNM but not close it in case the other guy falls through, in which case I'll take this one.

**Comment by Mothblocks** [comment]
I'm closing this, but feel free to ping me if the other PR doesn't look like it's getting done so I can reopen it.

**Pull request closed without merging**